src/core/db/repository/user.py

```diff
@@ -22,4 +23,40 @@ async def create_or_update(self, instance: User) -> User | None:
         db_instance = await self.get_by_username(instance.username)
         if db_instance is None:
             return await self.create(instance)
-        return await self.update(db_instance.id, instance)
+        return await self.update(instance.id, db_instance)
+
+    async def get_by_status(self, status: str) -> list[User] | None:
+        """Получает пользователей по статусу участия во встречах."""
+        async with self._sessionmaker() as session:
+            return await session.scalars(select(self._model).where(self._model.status == status))
+
+    async def get_free_user(self) -> User | None:
+        """Получает пользователя ожидающего встречи."""
+        async with self._sessionmaker() as session:
+            return await session.scalar(
+                select(self._model)
+                .options(joinedload(self._model.matches))
+                .where(self._model.status == StatusEnum.WAITING_MEETING)
+                .limit(1)
+            )
+
+    async def get_without_current(self, user: User) -> User | None:
+        """Возвращает пользователя со статусом ожидания встречи, исключая текущего пользователя."""
+        async with self._sessionmaker() as session:
+            return await session.scalar(
+                select(self._model)
+                .options(joinedload(self._model.matches))
+                .where(self._model.id != user.id, self._model.status == StatusEnum.WAITING_MEETING)
+                .join(self._model.matches)
+                .filter(
```

Comment:
Мы по всему проекту используем `.where()`

```suggestion
                .where(
```